gtsfm/common/constraint.py

```diff
@@ -0,0 +1,79 @@
+"""A pose constraint between two poses.
+
+Author: Frank Dellaert
+"""
+
+from dataclasses import dataclass
+from typing import List
+
+import gtsam
+import numpy as np
+
+
+@dataclass
+class Constraint:
+    """A pose constraint between two poses.
+    a and b are the pose indices.
+    aTb and cov are the relative pose and 6x6 covariance on tangent space.
```

Comment:
Do we actually have non diagonal entries on the pose covariances? This is because I was using diagonal noise models in my other PRs